src/SourceLink.Common/build/Microsoft.SourceLink.Common.targets

```diff
@@ -42,6 +55,13 @@
     <ItemGroup>
       <FileWrites Include="$(SourceLink)" />
     </ItemGroup>
+
+    <!-- C++ Link task currently doesn't recognize SourceLink property -->
```

Comment:
I think this block may solve https://stackoverflow.com/q/52423484/23059

Reply:
I'm doing it like this - defining custom cpp.props file:

    <?xml version="1.0" encoding="utf-8"?>
    <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
        <ItemDefinitionGroup>
            <Link>
                <GenerateDebugInformation>DebugFull</GenerateDebugInformation>
                <AdditionalOptions>/SOURCELINK:"${sourcelink}" %(AdditionalOptions)</AdditionalOptions>
            </Link>
        </ItemDefinitionGroup>
    </Project>
and then adding it to msbuild by /p:ForceImportBeforeCppTargets=cpp.props

will try your way now.

Reply:
Based on yours example I created custom target:

    <Target Name="IncludeSourceLinkFile" BeforeTargets="PreBuildEvent">
        <Message Text="IncludeSourceLinkFile..." />
        <ItemGroup Condition="'$(Language)' == 'C++'">
            <Link Update="@(Link)">
                <AdditionalOptions>%(Link.AdditionalOptions) /sourcelink:"$(SourceLink)"</AdditionalOptions>
            </Link>
        </ItemGroup>
    </Target>

and then added it via msbuild by /p:ForceImportBeforeCppTargets=cpp.props, it is executed but possibly too early as it does not affect link arguments